dt-impact-tracer/src/main/java/edu/washington/cs/dt/impact/tools/detectors/FailingTestDetector.java

```diff
@@ -57,24 +60,38 @@ private void output(final Set<String> tests) throws IOException {
         Files.write(outputPath, String.join(System.lineSeparator(), allOutput).getBytes());
     }
 
-    private Set<String> notPassingTests() throws IOException {
-        final List<String> tests = Files.readAllLines(testList, Charset.defaultCharset());
+    public FailingTestDetector(String classpath, List<String> tests) {
+        super(new String[0]);
+        this.classpath = classpath;
+        this.tests = tests;
+    }
+
+    public void setTestList(List<String> tests) {
+        this.tests = tests;
+    }
+
+    public Set<String> notPassingTests() throws IOException {
         final Set<String> notPassingTests = new HashSet<>();
 
         ImpactMain.skipMissingTests = true;
 
         while (true) {
+            if (tests.isEmpty()) {
+                break;
+            }
             final TestExecResult result = new FixedOrderRunner(classpath, tests).run().getExecutionRecords().get(0);
 
             final int beforeSize = tests.size();
             notPassingTests.addAll(tests.stream().filter(t -> !result.getResult(t).result.equals(RESULT.PASS)).collect(Collectors.toList()));
+
             tests.removeAll(notPassingTests);
             final int afterSize = tests.size();
 
             if (beforeSize == afterSize) {
                 break;
             }
         }
+        System.out.println();
```

Comment:
Largely because FixedOrderRunner outputs a bunch of stuff and it clutters the output without the line break afterward. I have moved this to the AutoCleanerMain for now though since I guess not everyone who may use the FailingTestDetector would want the line break.